### ISSUE TITLE
Temperature invalid on AMD platform 

### ISSUE THREAD

**Issue by decryptedchaos** [HorlogeSkynet/archey4#107]
Archey4  is trying to report temp1_input   However,  on the AMD platform,  this is used for Tctl  (T_control temperature)  which is highly inaccurate measurement.  (often 10-20c higher than actual die temp)  Effectively it's a smoothing mechanism, which reduces temp hysteria and smooths cooling curves. 

Reading the Tdie  (core/die temperature) is much more accurate.

I was unable to find a config option to specify this,  quickly went over the temperature.py  which is seems IMO overly complicated.  and couldn't find a obvious place to modify this behavior.  

**NOTE: I'm already on the latest v4.13.3**

```json
{
   "k10temp-pci-00c3":{
      "Tctl":{
         "temp1_input": 42.000
      },
      "Tdie":{
         "temp2_input": 32.000
      }
   },
   "iwlwifi_1-virtual-0":{
      "temp1":{
ERROR: Can't get value of subfeature temp1_input: Can't read

      }
   }
}
```

**Comment by HorlogeSkynet** [comment]
Thanks for reporting this.

A way I see out of here would be by implementing an additional configuration option excluding specific "subfeatures" (in `lm-sensors` terms), based on their identifiers.

In your very case, you would be able to define something like this to get rid of your Ryzen tCTL value :

```js
{
	// ...
	"entries": [
		// ...
		{
			"type": "Temperature",
			"sensors_chipsets": ["k10temp-pci-00c3"],
			"sensors_excluded_subfeatures": ["Tctl"]
		},
		// ...
	],
	// ...
}
```

What do you think about this ?
But more importantly I guess, what makes you think `temperature.py` "[seem] overly complicated" ?
This project is (supposed to be) community-driven, so any comments/changes would be appreciated.

Thanks for your time, bye :wave:

**Comment by decryptedchaos** [comment]
> A way I see out of here would be by implementing an additional configuration option excluding specific "subfeatures" (in `lm-sensors` terms), based on their identifiers.

I think this would be an good solution, to filter out the output we don't need. 

as for the code,  python isn't my forte,   so it just seems convoluted too me,  which is possibly due  my minimal knowledge of python code.   but from a bash script perspective we could accomplish the same thing with a   one liner awk pipe without all the extra overhead.   

This is why i labeled it overly complicated,  but this is likely just a matter of the  "python" way.

**Comment by HorlogeSkynet** [comment]
> I think this would be an good solution, to filter out the output we don't need.

:ok_hand: I'l try to implement this soon.

> as for the code, python isn't my forte, so it just seems convoluted too me, which is possibly due my minimal knowledge of python code. but from a bash script perspective we could accomplish the same thing with a one liner awk pipe without all the extra overhead.

Sure, everything can be done with a Bash one-liner :joy:
The idea here is to support multiple platforms (hard/ugly to achieve in scripts anyhow...) and allow a little of configuration (same here). Anyway, I tried to K.I.S.S. it when possible...

**Comment by HorlogeSkynet** [comment]
Hey @decryptedchaos, please see #108 and tell me if it actually fits your case.
Closing here as PR is available. Bye :wave:

**Comment by HorlogeSkynet** [comment]
Hi @decryptedchaos, this has just been released as v4.13.4 !
Bye :wave:
